dotfiles.py

```diff
@@ -1,29 +1,52 @@
 #!/usr/bin/env python3.8
-"""Module docstring."""
+
 import filecmp
+from collections import namedtuple
+from os.path import exists
 from os.path import expanduser
 from shutil import copyfile
 from typing import List
-from typing import Tuple
+from typing import NamedTuple
 
 import click
 
-DOTFILES: List[Tuple[str, str]] = [("~/.tmux.conf", "tmux.conf"), ("~/.vimrc", "vimrc")]
+
+class Dotfile(NamedTuple):
+    system_path: str
+    repo_name: str
+
+
+DOTFILES: List[Dotfile] = [
+    Dotfile(system_path=expanduser("~/.tmux.conf"), repo_name="tmux.conf"),
+    Dotfile(system_path=expanduser("~/.vimrc"), repo_name="vimrc"),
+]
 
 
 @click.group()
 def main():
-    """Pass"""
+    pass
 
 
 @main.command()
 def deploy():
-    """Deploy the dotfiles."""
-    dotfiles: Tuple[str, str] = [(expanduser(df[0]), df[1]) for df in DOTFILES]
+    """Deploy DOTFILES to system.
+
+    If the `system_path` of the dotfile already exists and matches the repo version,
+    do nothing. Otherwise, copy the repo config to the system path, overwriting
+    anything that might be there.
 
-    for system_file, my_file in dotfiles:
-        click.secho(system_file)
-        if not filecmp.cmp(my_file, system_file):
-            copyfile(my_file, system_file)
+    """
+    for dotfile in DOTFILES:
+        if exists(dotfile.system_path):
+            if filecmp.cmp(
+                dotfile.repo_name, dotfile.system_path
```

Comment:
According to the the [Python typeshed](https://github.com/python/typeshed/) the [signature of `filecmp.cmp`](https://github.com/python/typeshed/blob/473044cf1f80709b90fb6ed36581936685271667/stdlib/2and3/filecmp.pyi#L11-L13) takes a [`PathLike`](https://docs.python.org/3/library/os.html#os.PathLike), so it _should_ be safe to just pass a [`pathlib.Path`](https://docs.python.org/3/library/pathlib.html#pathlib.Path) in directly instead of needing to convert it with `str(Path(...))`.

Reply:
Verified that this works! 🥳